python/pyiceberg/expressions/base.py

```diff
@@ -99,8 +93,12 @@ def ref(self) -> BoundReference[T]:
         return self
 
 
+class UnboundTerm(Term[T], Unbound[BoundTerm[T]], ABC):
```

Comment:
Nit / non-blocking: An `Unbound[BoundTerm[T]]` is somewhat of a confusing type signature (particularly coming more from the Java side).

However `UnboundTerm` itself is clear as day so that's not particularly a concern.